core/domain/exp_domain.py

```diff
@@ -1144,9 +1136,15 @@ def get_all_outcomes(self):
             outcomes.append(self.default_outcome)
         return outcomes
 
-    def validate(self):
+    def validate(self, exp_param_specs_dict):
         """Validates various properties of the InteractionInstance.
 
+        Args:
+            exp_param_specs_dict: dict. A dict of specified parameters used in
```

Comment:
shouldn't this be "dict or None", to account for the Question case?

Also maybe in the Question case it makes sense to pass an empty dict? Not sure.

Reply:
I changed the validate call in question.validate() to empty dict.